apps/backend/src/drink-action/drink-action.service.ts

```diff
@@ -1,25 +1,44 @@
-import { Injectable, NotFoundException } from '@nestjs/common';
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaClientKnownRequestError } from '@prisma/client/runtime/library';
 import { PrismaService } from 'nestjs-prisma';
 
 import { CreateDrinkActionDto } from './dto/create-drink-action.dto';
+import { DrinkActionDto } from './dto/drink-action.dto';
 import { UpdateDrinkActionDto } from './dto/update-drink-action.dto';
 import { DrinkAction } from './entities/drink-action.entity';
 
 @Injectable()
 export class DrinkActionsService {
   constructor(private prisma: PrismaService) {}
 
-  async create(createDrinkActionDto: CreateDrinkActionDto) {
-    return this.prisma.drinkAction.create({ data: createDrinkActionDto });
+  async create(createDrinkActionDto: CreateDrinkActionDto): Promise<DrinkAction> {
+    try {
+      const { drinkId, eventId, ...rest } = createDrinkActionDto;
+      return await this.prisma.drinkAction.create({
+        data: {
+          ...rest,
+          drink: { connect: { id: drinkId } },
+          event: { connect: { id: eventId } },
+        },
+        include: { drink: true, event: true },
```

Comment:
This is not necessary here. We don't usually do anything with the return object of a POST request, so there's no need to add extra joins. Returning just the standard DrinkAction is fine